packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/utils/diffRelations.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @param {Array<{id: string}>} browserStateRelations
+ * @param {Array<{id: string}>} serverStateRelations
+ * @returns {[connected: string[], disconnected: string[]]} – the connected and disconnected relations ids
+ */
+export const diffRelations = (browserStateRelations = [], serverStateRelations = []) => {
+  const currentRelations = browserStateRelations.map((relation) => relation.id);
+  const oldRelations = serverStateRelations.map((relation) => relation.id);
+
+  const connected = currentRelations.filter((relation) => !oldRelations.includes(relation));
+  const disconnected = oldRelations.filter((relation) => !currentRelations.includes(relation));
```

Comment:
Should we try and avoid looping twice here? I also find this easier to read ...

```suggestion
  const connected = browserStateRelations.filter((relation) => !oldRelations.includes(relation.id));
  const disconnected = serverStateRelations.filter((relation) => !currentRelations.includes(relation.id));
```